### PR TITLE
feat: add onClick props to Message component (#28131)

### PR DESCRIPTION
**Pull request by ZeroTo0ne**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#28131

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    add onClick method to Message component       |
| 🇨🇳 Chinese |  message 组件新增 onClick 方法   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

-----
[View rendered components/message/index.en-US.md](https://github.com/ZeroTo0ne/ant-design/blob/feature-message/components/message/index.en-US.md)
[View rendered components/message/index.zh-CN.md](https://github.com/ZeroTo0ne/ant-design/blob/feature-message/components/message/index.zh-CN.md)

### CODE REVIEW

File: components/message/index.en-US.md [hunk 1]
```diff
@@ -17,23 +17,25 @@ Display global messages as feedback in response to user operations.
...
+- `message.warning(content, [duration], onClose, onClick)`
+- `message.warn(content, [duration], onClose, onClick)` // alias of warning
+- `message.loading(content, [duration], onClose, onClick)`
 
 | Argument | Description | Type | Default |
 | --- | --- | --- | --- |
 | content | The content of the message | ReactNode \| config | - |
 | duration | Time(seconds) before auto-dismiss, don't dismiss if set to 0 | number | 1.5 |
 | onClose | Specify a function that will be called when the message is closed | function | - |
+| onClick | Specify a function that will be called when the message is clicked | function | - |
```
Comment (afc163) [inline components/message/index.en-US.md:32]: Don't add `onClick` to here, keep it in `message.xxx({ ... })` would be enough.

Comment (ZeroTo0ne) [reply components/message/index.en-US.md:32]: 🙋🏻‍♂️组件内的实现只要提供这 message.xxx({ ... }) 一种 onClick 就可以了是吗？

Comment (afc163) [reply components/message/index.en-US.md:32]: 对，`message.xxx(a, b, c)` 这种 API 参数一多没法扩展了，就不动了。

File: components/message/__tests__/index.test.js [hunk 2]
```diff
@@ -65,6 +65,17 @@ describe('message', () => {
...
   });
 
+  it('trigger onClick method', () => {
+    const onClick = jest.fn();
+    message.info({
+      onClick: onClick,
+      duration: 2,
+      content: 'message info'
+    })
+    setTimeout(onClick(), 1000)
```
Comment (afc163) [inline components/message/__tests__/index.test.js:75]: 这个测试没意义呀，手动调用了一下肯定触发了。

File: components/message/index.tsx [hunk 3]
```diff
@@ -129,6 +129,7 @@ export interface ArgsProps {
   key?: string | number;
   style?: React.CSSProperties;
   className?: string;
+  onClick?: () => void;
```
Comment (afc163) [inline components/message/index.tsx:132]: 有 event 的参数

File: components/message/__tests__/index.test.js (hunk 2 above)
Comment (ZeroTo0ne) [reply components/message/__tests__/index.test.js]: 🤦🏻‍♂️ 已修正新的测试用例...

**Review by afc163 — APPROVED** [review]

**Pull request merged**